Change events schema to reflect PES modularity changes and provided_data_streams

Wecros opened the pull request:
* Add "modulestreams" array to the in_packageset and out_packageset
* Add "modulestream_maps" field defining the upgrade path for Move event
* [Allow null type for out_packageset object](https://github.com/oamg/schema-test/pull/6/commits/9c64759eea9d6e83aeb8b94b729178b70b253611)

I tested the updated schema validator on the new data (only one example) and it passed.

Wecros committed:
Change events schema to reflect PES modularity changes

* Add "modulestreams" array to the in_packageset and out_packageset
* Add "modulestream_maps" field defining the upgrade path for Move event

Wecros commented:
@pirat89 @vinzenz @MichalHe Can you guys please take a look at this schema update? I tested the updated schema (also with the other MR adding initial_release) against the latest pes-events json file. It passed OK.

Wecros committed:
Improve the structure of the PES events schema

* Defined "modulestream" and "package" as objects that are referenced
  later in the schema itself to reduce repetition & improve readability.
* Fixed issue with "items" property. There shouldn't be square brackets
  in its value, it causes issue where only the first item is validated.

Wecros committed:
Allow null type for out_packageset object

* Schema failed for previous events without this alteration.

Wecros commented:
I added "$defs" that can be referenced as subschemas. "modulestream" and "package" are now in these subschemas, reducing repetition as requested. @vinzenz

pirat89 commented on pes-events-schema-test.json:
```diff
@@ -2,118 +2,154 @@
-  "items": [
-    {
-      "type": "object",
+  "$defs": {
```
@Wecros  it's better to keep `$defs` section in the bottom of the document than in the top. So people can jump right on the schema and search for defs when they are interested. but it's nitpitck.

Wecros replied:
Moved as requested 👍

pirat89 committed:
pes-events-schema: Fix the schema & add provided_data_streams

The original schema has not reflected correctly the data format
of the pes-events.json file that has been used for a long time.
The main object's level has been completely skipped in the schema.
Also initial release has not been validated correctly.

The schema is fixed now + some additional small changes have been
done to improve the readability.

Also added provided_data_streams that is now mandatory. This validates
correctly PES data introduced by following PR:
    https://github.com/oamg/leapp-repository/pull/1092

pirat89 commented:
@Wecros I've discovered several issues that made the schema invalid when applied to the used data files. I've done several improvements and also added several updates to be compatible with up-to-date changes.

pirat89 commented:
Tested on various examples and updates and seems to reflect current data state. Merging.

pirat89 commented:
@Wecros the `type` field accepts list of types, so I've replaced `oneOf` / `anyOf`. The use of `oneOf` and `anyOf` has the negative effect that it's harder to realize what exactly is wrong with the particular object. E.g. here is one of testing issues I created for `anyOf`:
```
{'os_name': 'RHEL', 'major_version': 7, 'z_stream': None, 'tag': None}: {'os_name': 'RHEL', 'major_version': 7, 'z_stream': None, 'tag': None} is not valid under any of the given schemas
```

in the particular case I've removed `minor_version` field which is easier to identify when `...Of` is not used:
```
{'os_name': 'RHEL', 'major_version': 8, 'z_stream': None, 'tag': None}: 'minor_version' is a required property
```

Wecros commented:
Thanks for the info, good to see it was merged
